libs/sdk-core/src/swap_in/swap.rs

```diff
@@ -581,892 +545,1100 @@ impl BTCReceiveSwap {
-    let tx_out: Vec<TxOut> = vec![TxOut {
-        value: confirmed_amount,
-        script_pubkey: btc_address.payload.script_pubkey(),
-    }];
-
-    // construct the transaction
-    let tx = Transaction {
-        version: 2,
-        lock_time: crate::bitcoin::PackedLockTime(lock_time),
-        input: txins,
-        output: tx_out,
-    };
+        // If all utxos were used for payment, that means the swap server hasn't claimed them yet.
+        // There are no pending utxos, so the swap has completed.
+        if unpaid_utxos.is_empty() {
+            return SwapStatus::Completed;
+        }
 
-    Ok(tx)
-}
+        let refundable_utxos = refundable_utxos(swap_info, chain_data, current_tip, address_type);
 
-/// Creating the refund transaction that is to be used by the user in case where the swap has
-/// expired.
-fn create_refund_tx(
-    utxos: AddressUtxos,
-    private_key: Vec<u8>,
-    to_address: String,
-    lock_delay: u32,
-    input_script: &Script,
-    sat_per_vbyte: u32,
-) -> Result<Vec<u8>> {
-    info!("creating refund tx sat_per_vbyte {}", sat_per_vbyte);
-
-    let mut tx = prepare_refund_tx(&utxos, to_address, lock_delay)?;
-
-    let tx_weight = compute_refund_tx_weight(&tx);
-    let fees = compute_tx_fee(tx_weight, sat_per_vbyte);
-
-    if fees >= tx.output[0].value {
-        return Err(anyhow!("Insufficient funds to pay fees"));
-    }
-    tx.output[0].value -= fees;
-
-    let scpt = Secp256k1::signing_only();
-
-    // go over all inputs and sign them
-    let mut signed_inputs: Vec<TxIn> = Vec::new();
-    for (index, input) in tx.input.iter().enumerate() {
-        let mut signer = SighashCache::new(&tx);
-        let sig = signer.segwit_signature_hash(
-            index,
-            input_script,
-            utxos.confirmed[index].value,
-            EcdsaSighashType::All,
-        )?;
-        let msg = Message::from_slice(&sig[..])?;
-        let secret_key = SecretKey::from_slice(private_key.as_slice())?;
-        let sig = scpt.sign_ecdsa(&msg, &secret_key);
-
-        let mut sigvec = sig.serialize_der().to_vec();
-        sigvec.push(EcdsaSighashType::All as u8);
-
-        let witness: Vec<Vec<u8>> = vec![sigvec, vec![], input_script.serialize()];
-
-        let mut signed_input = input.clone();
-        let w = Witness::from_vec(witness);
-        signed_input.witness = w;
-        signed_inputs.push(signed_input);
-    }
-    tx.input = signed_inputs;
+        // There are utxos left, but they are not refundable yet. Mark the status as 'Redeemed' in that case.
+        if refundable_utxos.is_empty() {
+            return SwapStatus::Redeemed;
+        }
 
-    //tx.output[0].value = confirmed_amount;
-    Ok(encode::serialize(&tx))
-}
+        // There are refundable utxos.
+        SwapStatus::Refundable
+    }
 
-#[cfg(test)]
-mod tests {
-    use std::time::{SystemTime, UNIX_EPOCH};
-    use std::{sync::Arc, vec};
+    fn calculate_status_without_chain_data(
```

Comment:
In case use sent above the limit, should we consider the swap as refundable immediately? Currently it seems to be refundable only after the timeout?

Reply:
That makes sense for taproot swaps, yes. Will do.

Reply:
Done in fdb402b4538b269ece53ed115e6748b4d8e808d6